frontend/src/components/common/Terminal.tsx

```diff
@@ -321,32 +417,34 @@ export default function Terminal(props: TerminalProps) {
 
       return function cleanup() {
         xtermRef.current?.xterm.dispose();
-        execOrAttachRef.current?.cancel();
+        streamRef.current?.cancel();
         window.removeEventListener('resize', handler);
       };
     },
     // eslint-disable-next-line react-hooks/exhaustive-deps
     [container, terminalContainerRef, shells, props.open]
   );
 
-  React.useEffect(
-    () => {
-      if (props.open && container === null) {
-        setContainer(getDefaultContainer());
+  if (isPod(item)) {
+    React.useEffect(
```

Comment:
Hooks aren't supposed to be run conditionally as this leads to problems.

Reply:
This was addressed. Resolving.